.github/workflows/code-coverage.yaml

```diff
@@ -0,0 +1,72 @@
+name: code-coverage
+
+on:
+  push:
+    branches:
+      - development
+      - main
+  workflow_dispatch:
+  pull_request:
```

Comment:
github wofklow question from a newbie - does this mean we won't see the results of the unit tests when a PR is opened? only when merged?

also: is "code-coverage.yaml" the best name for this workflow? It seems like it's doing more than that. Maybe fortran-unit-tests.yaml?

Reply:
This will run the workflow when the PR is opened or updated (There's a series of workflow runs from my commits on this that should be visible under the `Checks` tab and there is drop down next to the commit sha that has the results of all the workflows from the initial creation of the PR).

As for the yaml file name, I'm amenable to anything that makes sense.  At the tail end of this, it is doing a code coverage run which requires the unit tests to be built but the naming conventions I've seen online are all over the place (some even just use `test.yaml` to do all testing/coverage/linting).

Reply:
ah yes, that makes sense.

in terms of the workflow name, i'm fine with `test.yaml` assuming that we're going to keep everything in one workflow rather than split them up like in CAM-SIMA

Reply:
So after a little more digging, `test.yaml` might be a little too generic (as there's other tests we'll want to add down the line aside from unit tests) so I went with `unit-tests.yaml` at the moment but if you think it should be more explicit or generic, happy to change it!